test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -54,6 +54,24 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForEnti
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData(false, "http://localhost/MyOrders(42)/OrderLines")]
```

Comment:
>http://localhost/MyOrders(42)/OrderLines [](start = 28, length = 40)

what if you have a single-value, non-contained, navigation source in the path.  i.e., assuming a Customer has a non-contained single-valued navigation property "mostRecentOrder" and the request was:
[http://localhost/Customers(42)/mostRecentOrder/OrderLines](http://localhost/Customers(42)/mostRecentOrder/OrderLines)